src/Bicep.LangServer/Completions/BicepCompletionContext.cs

```diff
@@ -551,6 +552,18 @@ private static bool IsArrayItemContext(List<SyntaxBase> matchingNodes, (ArraySyn
             // import foo from f|
             SyntaxMatcher.IsTailMatch<ImportDeclarationSyntax, IdentifierSyntax, Token>(matchingNodes, (import, ident, _) => import.ProviderName == ident);
 
+        private static bool IsSymbolicNameItemContext(List<SyntaxBase> matchingNodes, int offset) =>
+            // dependsOn on module
+            SyntaxMatcher.IsTailMatch<ModuleDeclarationSyntax, ObjectSyntax, ObjectPropertySyntax, ArraySyntax, Token>(matchingNodes, (_, _, objPropSyntax, _, _) => string.Equals(objPropSyntax.TryGetKeyText(), LanguageConstants.ResourceDependsOnPropertyName)) ||
+            // module1 = { scope: | }
+            SyntaxMatcher.IsTailMatch<ModuleDeclarationSyntax, ObjectSyntax, ObjectPropertySyntax>(matchingNodes, (_, _, objPropSyntax) => string.Equals(objPropSyntax.TryGetKeyText(), LanguageConstants.ResourceScopePropertyName)) ||
+            // dependsOn on resource
+            SyntaxMatcher.IsTailMatch<ResourceDeclarationSyntax, ObjectSyntax, ObjectPropertySyntax, ArraySyntax, Token>(matchingNodes, (_, _, objPropSyntax, _, _) => string.Equals(objPropSyntax.TryGetKeyText(), LanguageConstants.ResourceDependsOnPropertyName)) ||
```

Comment:
>ObjectSyntax [](http://example.com/codeflow?start=65&length=12)

This won't match module/resource loops or conditional loops. Sorry I didn't catch this early, but I'm now wondering if we should rely on the type of the property in context to filter out completions that are offered rather than a new context kind. (The `dependsOn`, `parent`, and `scope` properties have very specific types.)